myapp/spec/models/user_spec.rb

```diff
@@ -0,0 +1,40 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id              :integer          unsigned, not null, primary key
+#  email           :string(255)      not null
+#  is_admin        :boolean          default(FALSE), not null
+#  name            :string(255)      not null
+#  password_digest :string(255)      not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+# Indexes
+#
+#  index_users_on_email  (email) UNIQUE
+#
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_presence_of(:email) }
+
+    it 'duplicated email is not allowed' do
```

Comment:
Do you want to check only `email` or `user` ?

Reply:
Not sure I got your question, here I wanna check if user with a duplicated `email` is invalid 👀

Reply:
@anneju 
I think you should use only same email if you want to check the duplicated `email`.
`dup` copies every record.

Reply:
ah, okay. I will make it clearer.